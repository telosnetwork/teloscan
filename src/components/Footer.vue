<template>
<div class="c-footer row">
    <div class="col-12">
        <router-link to="/">
            Teloscan
        </router-link>
        &copy;&nbsp;{{ year }}
    </div>
</div>
</template>

<script>
export default {
    name: 'AppFooter',
    data: () => ({
        year: (new Date()).getFullYear(),
    }),
};
</script>

<style lang="scss">
body.body--light .c-footer {
    color: $white
}
.c-footer {
    margin: 8px auto 24px;
    padding: 0 16px;
    max-width: 1200px;
}
</style>
