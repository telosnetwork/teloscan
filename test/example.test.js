import { mount } from '@vue/test-utils'

import Footer from 'components/Footer';

/*
    TODO https://github.com/telosnetwork/teloscan/issues/203
    
    add to docs:
    - snapshot testing
    - conventions for router / other plugins
    - test naming & nesting
    - directory structure of test directory
 */

describe('Example test', () => {
    it('should be able to find the Footer component', () => {
        const wrapper = mount(Footer, {
            global: {
                plugins: ['router'],
            },
        });

        expect(wrapper.element).toMatchSnapshot()
    });
});
