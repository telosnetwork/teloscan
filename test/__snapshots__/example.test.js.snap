// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example test should be able to find the Footer component 1`] = `
<div
  class="c-footer row"
>
  <div
    class="col-12"
  >
    <router-link
      to="/"
    >
       Teloscan 
    </router-link>
     © 2022
  </div>
</div>
`;
